src/libraries/System.Text.Json/tests/System.Text.Json.SourceGeneration.Tests/System.Text.Json.SourceGeneration.Tests.targets

```diff
@@ -86,6 +86,17 @@
     <Compile Include="TestClasses.CustomConverters.cs" />
   </ItemGroup>
 
+  <ItemGroup>
+    <ProjectReference Include="..\..\src\System.Text.Json.csproj" />
+    <ProjectReference Include="$(LibrariesProjectRoot)System.Collections.Immutable\src\System.Collections.Immutable.csproj" />
+  </ItemGroup>
+
+  <ItemGroup Condition="'$(TargetFramework)' != '$(NetCoreAppCurrent)'">
```

Comment:
Why are these 3 files needed?

Reply:
I'm getting following errors without it:

```
d:\src\runtime-2\src\libraries\System.Text.Json\tests\Common\UnsupportedTypesTests.cs(14,57): error CS0518: Predefined type 'System.Runtime.CompilerServices.IsExternalInit' is not defined or imported [d:\src\runtime-2\src\libraries\System.Text.Json\tests\System.Text.Json.SourceGeneration.Tests\System.Text.Json.SourceGeneration.Roslyn3.11.Tests.csproj]
d:\src\runtime-2\src\libraries\System.Text.Json\tests\System.Text.Json.SourceGeneration.Tests\JsonSerializerContextTests.cs(86,39): error CS0518: Predefined type 'System.Runtime.CompilerServices.IsExternalInit' is not defined or imported [d:\src\runtime-2\src\libraries\System.Text.Json\tests\System.Text.Json.SourceGeneration.Tests\System.Text.Json.SourceGeneration.Roslyn3.11.Tests.csproj]
d:\src\runtime-2\src\libraries\System.Text.Json\tests\System.Text.Json.SourceGeneration.Tests\JsonSerializerContextTests.cs(86,57): error CS0518: Predefined type 'System.Runtime.CompilerServices.IsExternalInit' is not defined or imported [d:\src\runtime-2\src\libraries\System.Text.Json\tests\System.Text.Json.SourceGeneration.Tests\System.Text.Json.SourceGeneration.Roslyn3.11.Tests.csproj]
d:\src\runtime-2\src\libraries\System.Text.Json\tests\Common\ConstructorTests\ConstructorTests.ParameterMatching.cs(502,10): error CS0246: The type or namespace name 'DynamicDependencyAttribute' could not be found (are you missing a using directive or an assembly reference?) [d:\src\runtime-2\src\libraries\System.Text.Json\tests\System.Text.Json.SourceGeneration.Tests\System.Text.Json.SourceGeneration.Roslyn3.11.Tests.csproj]
d:\src\runtime-2\src\libraries\System.Text.Json\tests\Common\ConstructorTests\ConstructorTests.ParameterMatching.cs(502,10): error CS0246: The type or namespace name 'DynamicDependency' could not be found (are you missing a using directive or an assembly reference?) [d:\src\runtime-2\src\libraries\System.Text.Json\tests\System.Text.Json.SourceGeneration.Tests\System.Text.Json.SourceGeneration.Roslyn3.11.Tests.csproj]
d:\src\runtime-2\src\libraries\System.Text.Json\tests\Common\ConstructorTests\ConstructorTests.ParameterMatching.cs(502,28): error CS0122: 'DynamicallyAccessedMemberTypes' is inaccessible due to its protection level [d:\src\runtime-2\src\libraries\System.Text.Json\tests\System.Text.Json.SourceGeneration.Tests\System.Text.Json.SourceGeneration.Roslyn3.11.Tests.csproj]
d:\src\runtime-2\src\libraries\System.Text.Json\tests\Common\ConstructorTests\ConstructorTests.ParameterMatching.cs(502,80): error CS0122: 'DynamicallyAccessedMemberTypes' is inaccessible due to its protection level [d:\src\runtime-2\src\libraries\System.Text.Json\tests\System.Text.Json.SourceGeneration.Tests\System.Text.Json.SourceGeneration.Roslyn3.11.Tests.csproj]
d:\src\runtime-2\src\libraries\System.Text.Json\tests\Common\ConstructorTests\ConstructorTests.ParameterMatching.cs(529,10): error CS0246: The type or namespace name 'DynamicDependencyAttribute' could not be found (are you missing a using directive or an assembly reference?) [d:\src\runtime-2\src\libraries\System.Text.Json\tests\System.Text.Json.SourceGeneration.Tests\System.Text.Json.SourceGeneration.Roslyn3.11.Tests.csproj]
d:\src\runtime-2\src\libraries\System.Text.Json\tests\Common\ConstructorTests\ConstructorTests.ParameterMatching.cs(529,10): error CS0246: The type or namespace name 'DynamicDependency' could not be found (are you missing a using directive or an assembly reference?) [d:\src\runtime-2\src\libraries\System.Text.Json\tests\System.Text.Json.SourceGeneration.Tests\System.Text.Json.SourceGeneration.Roslyn3.11.Tests.csproj]
d:\src\runtime-2\src\libraries\System.Text.Json\tests\Common\ConstructorTests\ConstructorTests.ParameterMatching.cs(529,28): error CS0122: 'DynamicallyAccessedMemberTypes' is inaccessible due to its protection level [d:\src\runtime-2\src\libraries\System.Text.Json\tests\System.Text.Json.SourceGeneration.Tests\System.Text.Json.SourceGeneration.Roslyn3.11.Tests.csproj]
d:\src\runtime-2\src\libraries\System.Text.Json\tests\Common\ConstructorTests\ConstructorTests.ParameterMatching.cs(529,80): error CS0122: 'DynamicallyAccessedMemberTypes' is inaccessible due to its protection level [d:\src\runtime-2\src\libraries\System.Text.Json\tests\System.Text.Json.SourceGeneration.Tests\System.Text.Json.SourceGeneration.Roslyn3.11.Tests.csproj]
d:\src\runtime-2\src\libraries\System.Text.Json\tests\Common\ConstructorTests\ConstructorTests.ParameterMatching.cs(1087,36): error CS0518: Predefined type 'System.Runtime.CompilerServices.IsExternalInit' is not defined or imported [d:\src\runtime-2\src\libraries\System.Text.Json\tests\System.Text.Json.SourceGeneration.Tests\System.Text.Json.SourceGeneration.Roslyn3.11.Tests.csproj]
d:\src\runtime-2\src\libraries\System.Text.Json\tests\Common\ConstructorTests\ConstructorTests.ParameterMatching.cs(1100,37): error CS0518: Predefined type 'System.Runtime.CompilerServices.IsExternalInit' is not defined or imported [d:\src\runtime-2\src\libraries\System.Text.Json\tests\System.Text.Json.SourceGeneration.Tests\System.Text.Json.SourceGeneration.Roslyn3.11.Tests.csproj]
d:\src\runtime-2\src\libraries\System.Text.Json\tests\Common\ConstructorTests\ConstructorTests.ParameterMatching.cs(1112,59): error CS0518: Predefined type 'System.Runtime.CompilerServices.IsExternalInit' is not defined or imported [d:\src\runtime-2\src\libraries\System.Text.Json\tests\System.Text.Json.SourceGeneration.Tests\System.Text.Json.SourceGeneration.Roslyn3.11.Tests.csproj]
d:\src\runtime-2\src\libraries\System.Text.Json\tests\Common\ConstructorTests\ConstructorTests.ParameterMatching.cs(1112,73): error CS0518: Predefined type 'System.Runtime.CompilerServices.IsExternalInit' is not defined or imported [d:\src\runtime-2\src\libraries\System.Text.Json\tests\System.Text.Json.SourceGeneration.Tests\System.Text.Json.SourceGeneration.Roslyn3.11.Tests.csproj]
d:\src\runtime-2\src\libraries\System.Text.Json\tests\Common\PropertyVisibilityTests.InitOnly.cs(54,37): error CS0518: Predefined type 'System.Runtime.CompilerServices.IsExternalInit' is not defined or imported [d:\src\runtime-2\src\libraries\System.Text.Json\tests\System.Text.Json.SourceGeneration.Tests\System.Text.Json.SourceGeneration.Roslyn3.11.Tests.csproj]
d:\src\runtime-2\src\libraries\System.Text.Json\tests\Common\PropertyVisibilityTests.InitOnly.cs(59,37): error CS0518: Predefined type 'System.Runtime.CompilerServices.IsExternalInit' is not defined or imported [d:\src\runtime-2\src\libraries\System.Text.Json\tests\System.Text.Json.SourceGeneration.Tests\System.Text.Json.SourceGeneration.Roslyn3.11.Tests.csproj]
d:\src\runtime-2\src\libraries\System.Text.Json\tests\Common\PropertyVisibilityTests.InitOnly.cs(74,47): error CS0518: Predefined type 'System.Runtime.CompilerServices.IsExternalInit' is not defined or imported [d:\src\runtime-2\src\libraries\System.Text.Json\tests\System.Text.Json.SourceGeneration.Tests\System.Text.Json.SourceGeneration.Roslyn3.11.Tests.csproj]
d:\src\runtime-2\src\libraries\System.Text.Json\tests\Common\PropertyVisibilityTests.InitOnly.cs(92,47): error CS0518: Predefined type 'System.Runtime.CompilerServices.IsExternalInit' is not defined or imported [d:\src\runtime-2\src\libraries\System.Text.Json\tests\System.Text.Json.SourceGeneration.Tests\System.Text.Json.SourceGeneration.Roslyn3.11.Tests.csproj]
d:\src\runtime-2\src\libraries\System.Text.Json\tests\Common\PropertyVisibilityTests.NonPublicAccessors.cs(396,46): error CS0518: Predefined type 'System.Runtime.CompilerServices.IsExternalInit' is not defined or imported [d:\src\runtime-2\src\libraries\System.Text.Json\tests\System.Text.Json.SourceGeneration.Tests\System.Text.Json.SourceGeneration.Roslyn3.11.Tests.csproj]
d:\src\runtime-2\src\libraries\System.Text.Json\tests\Common\PropertyVisibilityTests.InitOnly.cs(64,45): error CS0518: Predefined type 'System.Runtime.CompilerServices.IsExternalInit' is not defined or imported [d:\src\runtime-2\src\libraries\System.Text.Json\tests\System.Text.Json.SourceGeneration.Tests\System.Text.Json.SourceGeneration.Roslyn3.11.Tests.csproj]
d:\src\runtime-2\src\libraries\System.Text.Json\tests\Common\PropertyVisibilityTests.InitOnly.cs(69,46): error CS0518: Predefined type 'System.Runtime.CompilerServices.IsExternalInit' is not defined or imported [d:\src\runtime-2\src\libraries\System.Text.Json\tests\System.Text.Json.SourceGeneration.Tests\System.Text.Json.SourceGeneration.Roslyn3.11.Tests.csproj]
d:\src\runtime-2\src\libraries\System.Text.Json\tests\Common\PropertyVisibilityTests.InitOnly.cs(80,45): error CS0518: Predefined type 'System.Runtime.CompilerServices.IsExternalInit' is not defined or imported [d:\src\runtime-2\src\libraries\System.Text.Json\tests\System.Text.Json.SourceGeneration.Tests\System.Text.Json.SourceGeneration.Roslyn3.11.Tests.csproj]
d:\src\runtime-2\src\libraries\System.Text.Json\tests\Common\PropertyVisibilityTests.InitOnly.cs(86,46): error CS0518: Predefined type 'System.Runtime.CompilerServices.IsExternalInit' is not defined or imported [d:\src\runtime-2\src\libraries\System.Text.Json\tests\System.Text.Json.SourceGeneration.Tests\System.Text.Json.SourceGeneration.Roslyn3.11.Tests.csproj]
d:\src\runtime-2\src\libraries\System.Text.Json\tests\Common\PropertyVisibilityTests.NonPublicAccessors.cs(384,44): error CS0518: Predefined type 'System.Runtime.CompilerServices.IsExternalInit' is not defined or imported [d:\src\runtime-2\src\libraries\System.Text.Json\tests\System.Text.Json.SourceGeneration.Tests\System.Text.Json.SourceGeneration.Roslyn3.11.Tests.csproj]
d:\src\runtime-2\src\libraries\System.Text.Json\tests\Common\PropertyVisibilityTests.NonPublicAccessors.cs(390,45): error CS0518: Predefined type 'System.Runtime.CompilerServices.IsExternalInit' is not defined or imported [d:\src\runtime-2\src\libraries\System.Text.Json\tests\System.Text.Json.SourceGeneration.Tests\System.Text.Json.SourceGeneration.Roslyn3.11.Tests.csproj]
```

Reply:
IsExternalInit is related to: `public int MyInt { get; private init; }` and similar
the rest seem to actually use the attribute

Reply:
IsExternalInit case seems weird I don't have enough knowledge of this - possibly a compiler bug? (it should generate this class if it's not defined I think - assuming compiler emits that attribute)

Reply:
cc: @jaredpar

Reply:
I thought @ericstj fixed the IsExternalInit issue with https://github.com/dotnet/runtime/pull/63472. Is that not the case?

Reply:
I fixed consumption in .NETCore when a netstandard library derives from S.T.J types with `init` properties.

This is a different scenario.  This is a netfx library which itself is trying to use `init` properties.  Much like S.T.J it needs to define that type for the compiler to permit use of the feature.

Reply:
FWIW we're doing the same in the regular reflection-based tests.  `IsExternalInit` still needs to be included for non-netcoreapp cases: https://github.com/dotnet/runtime/blob/fe627f5f8df996964fef3571d9e995e058a463fb/src/libraries/System.Text.Json/tests/System.Text.Json.Tests/System.Text.Json.Tests.csproj#L202-L206.